dace/transformation/interstate/loop_to_map.py

```diff
@@ -44,18 +45,47 @@ def can_be_applied(graph, candidate, expr_index, sdfg, strict=False):
             return False
 
         # If loop cannot be detected, fail
-        _, rng = find_for_loop(graph, guard, begin)
-        if not rng:
+        found = find_for_loop(graph, guard, begin)
+        if not found:
             return False
 
+        itervar, (start, end, step) = found
+
+        if step < 0:
```

Comment:
```suggestion
        if (step < 0) == True:
```
remember it's sympy

Reply:
you can still just invert the start/end and negate the step but I approve.

Reply:
c291450079cf53c692781f5c0094858f02ce8afd supports negative step